polyfill/spatial-navigation-polyfill.js

```diff
@@ -1059,62 +1083,111 @@
    * @returns {Points} The exit point from the search origin and the entry point from a candidate
    */
   function getEntryAndExitPoints(dir = 'down', rect1, rect2) {
-    const points = {entryPoint:[0,0], exitPoint:[0,0]};
+    const points = {entryPoint: {x: 0, y: 0}, exitPoint:{x: 0, y: 0}};
 
-    // Set direction
-    switch (dir) {
-    case 'left':
-      points.exitPoint[0] = rect1.left;
-      points.entryPoint[0] = (rect2.right < rect1.left) ? rect2.right : rect1.left;
-      break;
-    case 'up':
-      points.exitPoint[1] = rect1.top;
-      points.entryPoint[1] = (rect2.bottom < rect1.top) ? rect2.bottom : rect1.top;
-      break;
-    case 'right':
-      points.exitPoint[0] = rect1.right;
-      points.entryPoint[0] = (rect2.left > rect1.right) ? rect2.left : rect1.right;
-      break;
-    case 'down':
-      points.exitPoint[1] = rect1.bottom;
-      points.entryPoint[1] = (rect2.top > rect1.bottom) ? rect2.top : rect1.bottom;
-      break;
-    }
+    if (startingPoint) {
+      points.exitPoint = startingPoint;
 
-    // Set orthogonal direction
-    switch (dir) {
-    case 'left':
-    case 'right':
-      if (isBelow(rect1, rect2)) {
-        points.exitPoint[1] = rect1.top;
-        points.entryPoint[1] = (rect2.bottom < rect1.top) ? rect2.bottom : rect1.top;
-      }
-      else if (isBelow(rect2, rect1)) {
-        points.exitPoint[1] = rect1.bottom;
-        points.entryPoint[1] = (rect2.top > rect1.bottom) ? rect2.top : rect1.bottom;
-      }
-      else {
-        points.exitPoint[1] = Math.max(rect1.top, rect2.top);
-        points.entryPoint[1] = points.exitPoint[1];
+      switch (dir) {
+      case 'left':
+        points.entryPoint.x = (startingPoint.x > rect2.right) ? rect2.right : rect2.left;
+        break;
+      case 'up':
+        points.entryPoint.y = (startingPoint.y > rect2.bottom) ? rect2.bottom : rect2.top;
+        break;
+      case 'right':
+        points.entryPoint.x = (startingPoint.x < rect2.left) ? rect2.left : rect2.right;
+        break;
+      case 'down':
+        points.entryPoint.y = (startingPoint.y < rect2.top) ? rect2.top : rect2.bottom;
+        break;
       }
-      break;
-
-    case 'up':
-    case 'down':
-      if (isRightSide(rect1, rect2)) {
-        points.exitPoint[0] = rect1.left;
-        points.entryPoint[0] = (rect2.right < rect1.left) ? rect2.right : rect1.left;
+  
+      // Set orthogonal direction
+      switch (dir) {
+      case 'left':
+      case 'right':
+        if (startingPoint.y <= rect2.top) {
+          points.entryPoint.y = rect2.top;
+        }
+        else if (startingPoint.y < rect2.bottom) {
```

Comment:
if () {
} else if () {
} else {
}

Please unify the brace position

Reply:
Thanks.